The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/10508117987/job/29111327360) [(plain)](https://github.com/rust-lang/rust/commit/a9ecd0fa8141e7614ca35d5a4d2fe5c96b1a7890/checks/29111327360/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.833 Building wheels for collected packages: reuse
#13 2.834   Building wheel for reuse (pyproject.toml): started
#13 3.078   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.079   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 3.079   Stored in directory: /tmp/pip-ephem-wheel-cache-s1yjg7qs/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.082 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.490 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.490 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.023 Collecting virtualenv
#13 4.023 Collecting virtualenv
#13 4.071   Downloading virtualenv-20.26.3-py3-none-any.whl (5.7 MB)
#13 4.299      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 25.2 MB/s eta 0:00:00
#13 4.357 Collecting platformdirs<5,>=3.9.1
#13 4.365   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 4.388 Collecting distlib<1,>=0.3.7
#13 4.407      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 51.2 MB/s eta 0:00:00
#13 4.446 Collecting filelock<4,>=3.12.2
#13 4.453   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 4.453   Downloading filelock-3.15.4-py3-none-any.whl (16 kB)
#13 4.534 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.721 Successfully installed distlib-0.3.8 filelock-3.15.4 platformdirs-4.2.2 virtualenv-20.26.3
#13 DONE 4.8s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      202688 kB
DirectMap2M:     8185856 kB
DirectMap1G:    10485760 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/8269be147b13812c201d0b64966adc1c6536ca59/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-8269be147b13812c201d0b64966adc1c6536ca59-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 29.12s
##[endgroup]
fmt check
Diff in /checkout/library/std/src/os/unix/fs.rs:985:
 /// Changing the owner typically requires privileges, such as root or a specific capability.
 /// Changing the group typically requires either being the owner and a member of the group, or
 /// having privileges.
-/// 
+///
 /// Be aware that changing owner clears the `suid` and `sgid` permission bits in most cases
 /// according to POSIX, usually even if the user is root. The sgid is not cleared when
 /// the file is non-group-executable.
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/os/unix/net/datagram.rs" "/checkout/library/std/src/os/unix/net/ancillary.rs" "/checkout/library/std/src/os/unix/net/stream.rs" "/checkout/library/std/src/os/unix/net/tests.rs" "/checkout/library/std/src/os/unix/fs.rs" "/checkout/library/std/src/os/macos/mod.rs" "/checkout/library/std/src/os/net/mod.rs" "/checkout/library/std/src/os/net/linux_ext/socket.rs" "/checkout/library/std/src/os/net/linux_ext/mod.rs" "/checkout/library/std/src/os/net/linux_ext/addr.rs" "/checkout/library/std/src/os/net/linux_ext/tcp.rs" "/checkout/library/std/src/os/net/linux_ext/tests.rs" "/checkout/library/std/src/os/emscripten/raw.rs" "/checkout/library/std/src/os/emscripten/mod.rs" "/checkout/library/std/src/os/emscripten/fs.rs" "/checkout/library/std/src/os/ios/mod.rs" "/checkout/library/std/src/os/espidf/raw.rs" "/checkout/library/std/src/os/espidf/mod.rs" "/checkout/library/std/src/os/espidf/fs.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/main.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/utils.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/clean.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/config.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/clone_gcc.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/info.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/build.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/test.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/prepare.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/rustc_info.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/rust_tools.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/src/fmt.rs" "/checkout/library/std/src/os/fd/raw.rs" "/checkout/library/std/src/os/fd/mod.rs" "/checkout/library/std/src/os/fd/net.rs" "/checkout/library/std/src/os/fd/owned.rs" "/checkout/library/std/src/os/fd/tests.rs" "/checkout/library/std/src/path/tests.rs" "/checkout/library/std/src/sys/personality/emcc.rs" "/checkout/library/std/src/sys/personality/gcc.rs" "/checkout/library/std/src/sys/personality/mod.rs" "/checkout/library/std/src/sys/personality/dwarf/mod.rs" "/checkout/library/std/src/sys/personality/dwarf/eh.rs" "/checkout/library/std/src/sys/personality/dwarf/tests.rs" "/checkout/library/std/src/sys/sync/rwlock/teeos.rs" "/checkout/library/std/src/sys/sync/rwlock/queue.rs" "/checkout/library/std/src/sys/sync/rwlock/mod.rs" "/checkout/library/std/src/sys/sync/rwlock/no_threads.rs" "/checkout/library/std/src/sys/sync/rwlock/futex.rs" "/checkout/library/std/src/sys/sync/rwlock/solid.rs" "/checkout/library/std/src/sys/sync/mutex/sgx.rs" "/checkout/library/std/src/sys/sync/mutex/fuchsia.rs" "/checkout/library/std/src/sys/sync/mutex/mod.rs" "/checkout/library/std/src/sys/sync/mutex/no_threads.rs" "/checkout/library/std/src/sys/sync/mutex/futex.rs" "/checkout/library/std/src/sys/sync/mutex/pthread.rs" "/checkout/library/std/src/sys/sync/mutex/xous.rs" "/checkout/library/std/src/sys/sync/mutex/windows7.rs" "/checkout/library/std/src/sys/sync/mutex/itron.rs" "/checkout/library/std/src/sys/sync/mod.rs" "/checkout/compiler/rustc_codegen_gcc/src/errors.rs" "/checkout/compiler/rustc_codegen_gcc/src/base.rs" "/checkout/compiler/rustc_codegen_gcc/src/asm.rs" "/checkout/compiler/rustc_codegen_gcc/src/debuginfo.rs" "/checkout/compiler/rustc_codegen_gcc/build_system/build_sysroot/lib.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Thu Aug 22 12:37:44 UTC 2024
  network time: Thu, 22 Aug 2024 12:37:45 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/os/unix/fs.rs
+++ b/library/std/src/os/unix/fs.rs
@@ -988,7 +988,8 @@ impl DirBuilderExt for fs::DirBuilder {
 /// 
 /// Be aware that changing owner clears the `suid` and `sgid` permission bits in most cases
 /// according to POSIX, usually even if the user is root. The sgid is not cleared when
-/// the file is non-group-executable.
+/// the file is non-group-executable. See: <https://www.man7.org/linux/man-pages/man2/chown.2.html>
+/// This call may also clear file capabilities, if there was any.
 ///
 /// If called on a symbolic link, this will change the owner and group of the link target. To
 /// change the owner and group of the link itself, see [`lchown`].

